src/test/java/org/opensearch/security/auditlog/compliance/RestApiComplianceAuditlogTest.java

```diff
@@ -196,26 +214,45 @@ public void testBCryptHashRedaction() throws Exception {
             .put(ConfigConstants.SECURITY_COMPLIANCE_HISTORY_INTERNAL_CONFIG_ENABLED, true)
             .put(ConfigConstants.OPENDISTRO_SECURITY_COMPLIANCE_HISTORY_WRITE_LOG_DIFFS, true)
             .build();
-        setup(settings);
+        setupAndReturnAuditMessages(settings);
         rh.sendAdminCertificate = true;
         rh.keystore = "kirk-keystore.jks";
 
         // read internal users and verify no BCrypt hash is present in audit logs
-        TestAuditlogImpl.clear();
-        rh.executeGetRequest("/_opendistro/_security/api/internalusers");
-        Assert.assertEquals(1, TestAuditlogImpl.messages.size());
-        Assert.assertFalse(AuditMessage.BCRYPT_HASH.matcher(TestAuditlogImpl.sb.toString()).matches());
+        final AuditMessage message1 = TestAuditlogImpl.doThenWaitForMessage(() -> {
+            rh.executeGetRequest("/_opendistro/_security/api/internalusers");
+        });
+
+        Assert.assertFalse(AuditMessage.BCRYPT_HASH.matcher(message1.toString()).matches());
 
         // read internal user worf and verify no BCrypt hash is present in audit logs
-        TestAuditlogImpl.clear();
-        rh.executeGetRequest("/_opendistro/_security/api/internalusers/worf");
-        Assert.assertEquals(1, TestAuditlogImpl.messages.size());
-        Assert.assertFalse(AuditMessage.BCRYPT_HASH.matcher(TestAuditlogImpl.sb.toString()).matches());
+        final AuditMessage message2 = TestAuditlogImpl.doThenWaitForMessage(() -> {
+            rh.executeGetRequest("/_opendistro/_security/api/internalusers/worf");
+            Assert.assertFalse(AuditMessage.BCRYPT_HASH.matcher(TestAuditlogImpl.sb.toString()).matches());
+        });
+
+        Assert.assertFalse(AuditMessage.BCRYPT_HASH.matcher(message2.toString()).matches());
 
         // create internal user and verify no BCrypt hash is present in audit logs
-        TestAuditlogImpl.clear();
-        rh.executePutRequest("/_opendistro/_security/api/internalusers/test", "{ \"password\":\"some new user password\"}");
-        Assert.assertEquals(1, TestAuditlogImpl.messages.size());
-        Assert.assertFalse(AuditMessage.BCRYPT_HASH.matcher(TestAuditlogImpl.sb.toString()).matches());
+        final AuditMessage message3 = TestAuditlogImpl.doThenWaitForMessage(() -> {
+            rh.executePutRequest("/_opendistro/_security/api/internalusers/test", "{ \"password\":\"some new user password\"}");
+        });
+
+        Assert.assertFalse(AuditMessage.BCRYPT_HASH.matcher(message3.toString()).matches());
+    }
+
+    private List<AuditMessage> setupAndReturnAuditMessages(Settings settings) {
```

Comment:
Nice - this works for my concerns [1] raised 

- [1] https://github.com/opensearch-project/security/pull/3732#issuecomment-1816868686